File: 0496-next-greater-element-i/solution.java
class Solution {
    public int[] nextGreaterElement(int[] nums1, int[] nums2) {
        List<Integer> temp = new ArrayList<>();
        int[] res = new int[nums1.length];
        for(int i=0;i<nums2.length;i++){
            temp.add(nums2[i]);
        }

        for(int i=0;i<nums1.length;i++){
            int ind = temp.indexOf(nums1[i]);
            res[i]=check(nums2,ind,nums1[i]);
        }
        return res;
    }
    public int check(int[] nums2,int ind,int cur){
        int temp1=-1;
        for(int i=ind+1;i<nums2.length;i++){
            if(nums2[i]>cur){
                temp1=nums2[i];
                break;
            }
        }
        return temp1;
    }
}

File: 0496-next-greater-element-i/solution.cpp
class Solution {
public:
    vector<int> nextGreaterElement(vector<int>& nums1, vector<int>& nums2) {
        stack<int> st;
        unordered_map<int,int>m;
        for(auto val:nums2){
            while(!st.empty() && st.top() < val){
                m[st.top()]=val;
                st.pop();
            }
            st.push(val);
        }  
        while(!st.empty()){
            m[st.top()]=-1;
            st.pop();
        }
        vector<int> res;
        for(auto val:nums1){
            res.push_back(m[val]);
        }
        return res;
    }
};
